cmd/api/src/database/migration/migrations/v6.1.0.sql

```diff
@@ -1,13 +1,50 @@
+-- Copyright 2024 Specter Ops, Inc.
+--
+-- Licensed under the Apache License, Version 2.0
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+-- SSO Provider
+CREATE TABLE IF NOT EXISTS sso_providers
+(
+    id         SERIAL PRIMARY KEY,
+    name       TEXT NOT NULL,
+    slug       TEXT NOT NULL,
+    type       int  NOT NULL,
+
+    updated_at TIMESTAMP WITH TIME ZONE DEFAULT now(),
+    created_at TIMESTAMP WITH TIME ZONE DEFAULT now(),
+
+    UNIQUE (name),
+    UNIQUE (slug)
+);
+
 -- OIDC Provider
 CREATE TABLE IF NOT EXISTS oidc_providers
 (
-  id         BIGSERIAL PRIMARY KEY,
-  name       TEXT NOT NULL,
-  client_id  TEXT NOT NULL,
-  issuer     TEXT NOT NULL,
+    id              SERIAL PRIMARY KEY,
+    name            TEXT NOT NULL,
+    client_id       TEXT NOT NULL,
+    issuer          TEXT NOT NULL,
+    sso_provider_id INTEGER REFERENCES sso_providers (id) ON DELETE CASCADE NULL,
```

Comment:
This supports auto-deleting the oidc_provider row when the associated sso_provider is deleted.

We can handle the scenario when an oidc_provider is deleted, the associated sso_provider is deleted in:
https://github.com/SpecterOps/BloodHound/pull/890

Alternatively we could use a pg trigger to accomplish this, but I'd personally want to handle it through our application

Reply:
Agreed. This is a good use case for using `ON DELETE CASCADE` as opposed to a trigger.